plugins/tarpit.js

```diff
@@ -26,11 +26,14 @@ exports.tarpit = function (next, connection) {
 
     let delay = connection.notes.tarpit;
 
-    if (!delay && connection.transaction) {
+    if (delay == null && connection.transaction != null) {
```

Comment:
Here you are trading the imprecision of truthy and falsy with the imprecision of ==. Using == is an anti-pattern because so few people grok the difference between == and ===. Or so I thought.

Reply:
@msimerson I'm 100 % with you about the trade-off. 

I disagree although about people knowing the difference, I have been working on typescript on last years, so maybe I'm biased here.

I will stick to whichever looks most comfortable for Haraka's code base patterns.